README.md

```diff
@@ -291,43 +291,47 @@ As you can see, some of the properties — such as `metadata` — are loaded on
 - If you're only fetching one NFT — e.g. by using `findNftByMint` — then these properties will already be loaded.
 - If you're fetching multiple NFTs — e.g. by using `findNftsByMintLint` — then these properties will not be loaded and you will need to load them as and when you need them.
 
-In order to load these properties, you may use the `metadataLoader` and `masterEditionLoader` properties of the `Nft` object.
+In order to load these properties, you may run the `metadataTask` and `masterEditionTask` properties of the `Nft` object.
 
 ```ts
-await nft.metadataLoader.load();
-await nft.masterEditionLoader.load();
+await nft.metadataTask.run();
+await nft.masterEditionTask.run();
 ```
 
-After these two promises resolve, you should have access to the `metadata`, `masterEditionAccount` and `masterEdition` properties. Note that if a loader fails to load the data, an error will be thrown. You may change that behaviour by providing the `failSilently` option to the `load` method.
+After these two promises resolve, you should have access to the `metadata`, `masterEditionAccount` and `masterEdition` properties. Note that if a task fails to load the data, an error will be thrown.
 
-```ts
-await nft.metadataLoader.load({ failSilently: true });
-```
-
-Also, note that both `metadataLoader` and `masterEditionLoader` are instances of the `Loader` class which contains a bunch of helper methods. Here's an overview of the methods available in the `Loader` class:
+Also, note that both `metadataTask` and `masterEditionTask` are of type `Task` which contains a bunch of helper methods. Here's an overview of the methods available in the `Task` class:
 
 ```ts
-class Loader<T> {
-    public getStatus(): LoaderStatus;
-    public getResult(): T | undefined;
-    public getError(): unknown;
-    public isPending(): boolean;
-    public isLoading(): boolean;
-    public isLoaded(): boolean;
-    public isSuccessful(): boolean;
-    public isFailed(): boolean;
-    public isCanceled(): boolean;
+export type Task<T> = {
+    getStatus: () => TaskStatus;
+    getResult: () => T | undefined;
+    getError: () => unknown;
+    isPending: () => boolean;
+    isRunning: () => boolean;
+    isCompleted: () => boolean;
+    isSuccessful: () => boolean;
+    isFailed: () => boolean;
+    isCanceled: () => boolean;
+    run: (options?: TaskOptions) => Promise<T>;
+    loadWith: (preloadedResult: T) => Task<T>;
+    reset: () => Task<T>;
+    onStatusChange: (callback: (status: TaskStatus) => unknown) => Task<T>;
```

Comment:
If you're not consuming the return value of the callback (which you're probably not) it should have `void` return type instead of `unknown`

Reply:
I'm happy to go with `void` but let me explain to you why I went for `unknown` in the first place.

When using arrow functions for one-liners it immediately returns the evaluated expression. That means if you were to use the following listeners, TypeScript would give you a type error.

```ts
let taskStatus = 'pending';
task.onStatusChange(newStatus => taskStatus = newStatus) // Asignment returns the assigned value.

const statusHistory = ['pending'];
task.onStatusChange(newStatus => statusHistory.push(newStatus)) // Push returns the latest index.
```

Instead, you'd need to wrap your one-liner in curly braces for it to work with `void`.

```ts
task.onStatusChange(newStatus => { taskStatus = newStatus })
task.onStatusChange(newStatus => { statusHistory.push(newStatus) })
```

So all in all, not such a big difference but I did find the DX more enjoyable using `unknown`. That being said, I'm happy to go with `void` if you find that it will create less confusion. 🙂